lms/views/basic_lti_launch.py

```diff
@@ -113,14 +113,26 @@ def db_configured_basic_lti_launch(self):
         resource_link_id = self.request.params["resource_link_id"]
         tool_consumer_instance_guid = self.request.params["tool_consumer_instance_guid"]
 
-        # The ``db_configured=True`` view predicate ensures that this view
-        # won't be called if there isn't a matching ModuleItemConfiguration in
-        # the DB. So here we can safely assume that the ModuleItemConfiguration
-        # exists.
         document_url = ModuleItemConfiguration.get_document_url(
             self.request.db, tool_consumer_instance_guid, resource_link_id
         )
 
+        if document_url is None:
+            previous_resource_link_id = self.request.params.get(
+                "resource_link_id_history"
+            )
+
+            document_url = ModuleItemConfiguration.get_document_url(
+                self.request.db, tool_consumer_instance_guid, previous_resource_link_id
+            )
+
+            ModuleItemConfiguration.set_document_url(
+                self.request.db,
+                tool_consumer_instance_guid,
+                resource_link_id,
+                document_url,
+            )
```

Comment:
Basically what this does is, if we don't have an assignment configuration for this assignment's ID, then it looks for an assignment configuration for the `resource_link_id_history` (the ID of the original assignment that this assignment was copied from) instead and copies that assignment configuration to the new assignment's ID.

If we don't find an assignment configuration (`ModuleItemConfiguration`) for the `resource_link_id` launch param in our DB then we look for a `ModuleItemConfiguration` for the `resource_link_id_history` launch param instead. If we find one, then we copy its `document_url` into a new `ModuleItemConfiguration` for the `resource_link_id`.

It needs to be a bit more careful about the cases where there's no `resource_link_id_history` param, or where there is one but we don't find it in our DB.